docs/model-dev-guide/optimize-training.rst

```diff
@@ -0,0 +1,111 @@
+.. _profiling:
+
+###########
+ Profiling
+###########
+
+Optimizing a model's performance is often a trade-off between accuracy, time, and resource
+requirements. Training deep learning models is a time and resource intensive process, where each
+iteration can take several hours and accumulate heavy hardware costs. Though sometimes this cost is
+inherent to the task, unnecessary resource consumption can be caused by suboptimal code or bugs.
+Thus, achieving optimal model performance requires an understanding of how your model interacts with
+the system's computational resources.
+
+Profiling collects metrics on how computational resources like CPU, GPU, and memory are being
+utilized during a training job. It can reveal patterns in resource utilization that indicate
+performance bottlenecks and pinpoint areas of the code or pipeline that are causing slowdowns or
+inefficiencies.
+
+Profiling a training job can be instrumented at many different layers, from generic system-level
+metrics to individual model operators and GPU kernels. Determined provides a few options for
+profiling, each targeting a different layer in a training job at various levels of detail:
+
+-  :ref:`Determined system metrics profiler <how-to-profiling-det-profiler>` collects general
+   system-level metrics and provides an overview of hardware usage during an experiment.
+-  :ref:`Native profiler integration <how-to-profiling-native-profilers>` enables model profiling in
+   training APIs that provides fine-grained metrics specific to your model.
+-  :ref:`Prometheus/Grafana integration <how-to-profiling-prom-grafana>` can be set up to track
+   detailed hardware metrics and monitor overall cluster health.
+
+.. _how-to-profiling:
+
+.. _how-to-profiling-det-profiler:
+
+*********************
+ Determined Profiler
+*********************
+
+Determined comes with a built-in profiler that provides out-of-the-box tracking for system-level
+metrics. System metrics are statistics around hardware usage, such as GPU utilization, disk usage,
+and network throughput.
+
+These metrics provide a general overview of resource usage during a training run, and can be useful
+for quickly identifying ineffective usages of computational resources. When the system metrics
+reported for an experiment do not match hardware expectations, that is a sign that the software may
+be able to be optimized to make better use of the hardware resources.
+
+The Determined profiler collects a set of system metrics throughout an experiment which can be
+visualized in the Web UI under the experiment's "Profiler" tab. It is supported for all training
+APIs, but is not enabled by default.
+
+Please see :ref:`core-profiler` for details on enabling and configuring the Determined profiler for
+your experiment.
+
```

Comment:
Visit :ref:`core-profiler` to find out how to enable and configure the Determined profiler for
your experiment.